io-engine/src/prctl.rs

```diff
@@ -0,0 +1,23 @@
+use libc::c_int;
+use std::io::{Error, Result};
+
+const PR_SET_IO_FLUSHER: c_int = 57;
+
+pub struct Prctl;
+
+impl Prctl {
+    // From man page of prctl: If a user process is involved in the block layer
+    // or filesystem I/O path, and can allocate memory while processing I/O requests
+    // it must set arg2 to 1. This will put the process in the IO_FLUSHER  state,
+    // which allows it special treatment to make progress when allocating memory.
+    // If arg2 is 0, the process will clear the IO_FLUSHER state, and the default
+    // behavior will be used.
+
```

Comment:
Using 3 /// for doc comments

```suggestion
    /// From man page of prctl: If a user process is involved in the block layer
    /// or filesystem I/O path, and can allocate memory while processing I/O requests
    /// it must set arg2 to 1. This will put the process in the IO_FLUSHER  state,
    /// which allows it special treatment to make progress when allocating memory.
    /// If arg2 is 0, the process will clear the IO_FLUSHER state, and the default
    /// behavior will be used.
```

Reply:
I applied the suggested change locally, changed to pass pre-commit, and force pushed to keep the commit history simple.  The changes are visible using <https://github.com/daclash/mayastor/compare/44217aa0761e31660448e34c4ec515f0d3b005d9..e9c8fa4558d356eaa30168b0f3114f2d3d87da83>.